src/lib/telemetry.rs

```diff
@@ -136,12 +136,13 @@ pub enum LogKey {
 
 impl LogKey {
     pub fn add_suffix(&self, suffix: &str) -> LogKey {
-        let s = format!("{}-{}", &self.to_string(), suffix);
-        let s_str = s.as_str();
+        let mut s = self.to_string();
+        s.push_str("-");
+        s.push_str(suffix);
 
-        match LogKey::from_str(s_str) {
+        match LogKey::from_str(&s) {
             Ok(v) => v,
-            Err(_) => LogKey::from_str(&self.to_string()).unwrap(),
+            Err(_) => *self,
```

Comment:
Deferencing here isn't normally possible because this function owns this reference, so it must be dropped at the end and Rust will complain if you try to reuse it (I think). But you can do this if you derive `Copy`. Which is way better than whatever mess I came up with before.